### DISCUSSION TITLE
Traffic in/out bytes on client down

### DISCUSSION THREAD

**Discussion by realbitt** [strongswan/strongswan#1145] [Ideas]
I know that this topic has been brought up a lot, and I know the solutions offered.
I have read many threads on this matter.

The proposed solutions are to use **Radius Accounting**
In my case I need to install **Radius** on a separate server. The other thing is the problem of data transfer from radius server to main server which is running with windows server also NPS  SQL Logs not contains **Acc-Input/Output-Octets**  but txt logs contains these fields and its too hard to parse  text  logs to catch these values, All this can be solved with a simple solution, which is to provide in/out bytes data in the updown script and these information are available in this scope,  this solution saves a lot of time. And resources too.

I am trying to modify the original source to present these values ​​in the updown script .

After searching, I found here what I'm looking for.

```
src/libcharon/sa/child_sa.c
src/libcharon/plugins/updown/updown_listener.c

```
and insert similar code like this for **get_usestats**
```
	push_env(envp, countof(envp), "PLUTO_INTERFACE=%s",
			 iface ? iface : "unknown");

```
 
While reading the source code, I found this function that does the job **get_usestats** ,  but I'm not familiar with **C Lang** so I need some push ;)

Thanks

**Comment by tobiasbrunner** [comment]
The _updown_ plugin and script is deprecated. Use [vici](https://docs.strongswan.org/docs/5.9/plugins/vici.html) and you have all that stuff in the child-updown event (and the child-rekey event).
